src/data/finset.lean

```diff
@@ -365,6 +371,35 @@ by simp only [insert_eq, union_left_comm]
 theorem insert_union_distrib (a : α) (s t : finset α) : insert a (s ∪ t) = insert a s ∪ insert a t :=
 by simp only [insert_union, union_insert, insert_idem]
 
+@[simp] lemma union_eq_left_iff_subset {s t : finset α} :
+  s ∪ t = s ↔ t ⊆ s :=
+begin
+  split,
+  { assume h,
+    have : t ⊆ s ∪ t := subset_union_right _ _,
+    rwa h at this },
+  { assume h,
+    exact subset.antisymm (union_subset (subset.refl _) h) (subset_union_left _ _) }
+end
+
+@[simp] lemma left_eq_union_iff_subset {s t : finset α} :
+  s = s ∪ t ↔ t ⊆ s :=
+by rw [← union_eq_left_iff_subset, eq_comm]
+
+@[simp] lemma union_eq_right_iff_subset {s t : finset α} :
+  t ∪ s = s ↔ t ⊆ s :=
+begin
```

Comment:
Can't you golf this using `union_comm` and one of the preceding lemmas?

Reply:
Yes, thanks! Done.